www/data_service/src/services/data/data.service.coffee

```diff
@@ -0,0 +1,146 @@
+# TODO rename this, or the dataService in waterfall view
+class Data extends Provider
+    constructor: ->
+    # TODO caching
+    cache: false
+
+    $get: ($log, $injector, $q, restService, socketService, helperService, ENDPOINTS) ->
+        return new class DataService
```

Comment:
right, but then we have to write @$service, which is ugly, that's why I
started to use this type of construction

On Thu, Jun 11, 2015, 17:39 Pierre Tardy <notifications@github.com> wrote:

> In www/data_service/src/services/data/data.service.coffee
> <https://github.com/buildbot/buildbot/pull/1703#discussion_r32234407>:
>
> > @@ -0,0 +1,146 @@
> > +# TODO rename this, or the dataService in waterfall view
> > +class Data extends Provider
> > +    constructor: ->
> > +    # TODO caching
> > +    cache: false
> > +
> > +    $get: ($log, $injector, $q, restService, socketService, helperService, ENDPOINTS) ->
> > +        return new class DataService
>
> no, but we can pass it to the contructor
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/buildbot/buildbot/pull/1703/files#r32234407>.
>